tests/components/smarttub/test_light.py

```diff
@@ -0,0 +1,38 @@
+"""Test the SmartTub light platform."""
+
+from smarttub import SpaLight
+
+
+async def test_light(spa, setup_entry, hass):
+    """Test light entity."""
+
+    for light in spa.get_lights.return_value:
+        entity_id = f"light.{spa.brand}_{spa.model}_light_{light.zone}"
+        state = hass.states.get(entity_id)
+        assert state is not None
+        if light.mode == SpaLight.LightMode.OFF:
```

Comment:
It's better to use pytest parametrization instead of these checks so we can avoid the logic inside the test.

https://docs.pytest.org/en/stable/parametrize.html

Reply:
Was addressed here https://github.com/home-assistant/core/pull/47189